Examples/Algorithms/TruthTracking/ActsExamples/TruthTracking/ParticleSmearing.hpp

```diff
@@ -67,6 +67,8 @@ class ParticleSmearing final : public IAlgorithm {
         0.1 / Acts::UnitConstants::GeV,  1 * Acts::UnitConstants::ns};
     /// Inflate the initial covariance matrix
     std::array<double, 6> initialVarInflation = {1., 1., 1., 1., 1., 1.};
+    /// Add covariance to the generated parameters
+    bool addCovariances = true;
```

Comment:
Thinking about it I believe it would be cleaner to push this flag to the `PropagationAlgorithm` because this is the only case where we want to ignore the covariances I think?

Reply:
This is how it started, however, I get a trackParamters object and I would have to clone it without error - in ATLAS we have a method for this, but I don't really like it.

Reply:
Right but now we change the unrelated components which also conflicts with https://github.com/acts-project/acts/pull/3422

```
TrackParameters(parameter.referenceSurface().getSharedPtr(), parameter.parameters(), std::nullopt, parameter.particleHypothesis());
```
with switching behavior for the covariance does not seem to bad to me in the `PropagationAlgorithm`

Reply:
ok, will backe that in then, is fine for me.

Reply:
Done.